test/end2end.py

```diff
@@ -308,3 +309,14 @@ def change_env(self, cores):
         pbsnodefile.write('\n'.join(['localhost'] * cores))
         pbsnodefile.close()
         os.environ['PBS_NODEFILE'] = pbsnodefile.name
+
+
+    def test_unset_nodefile(self):
+        self.assertTrue(schedm.what_sched(False)[0].__name__ == 'PBS')
+        nodefile = os.environ['PBS_NODEFILE']
+        del os.environ['PBS_NODEFILE']
+        # fall back to local if PBS_NODEFILE is not available
+        self.assertTrue(schedm.what_sched(False)[0].__name__ == 'Local')
+
+        # reset nodefile for proper test cleanup
+        os.environ['PBS_NODEFILE'] = nodefile
```

Comment:
rather than using `cleanup_PBS_env`, we should just restore the original environment in `tearDown` using

```python
os.environ = self.orig_environ
```

so, let's trash `cleanup_PBS_env`?